### PR TITLE
remove uses of stdatamodels.jwst.datamodels.schema

### PR DESCRIPTION
**Pull request by braingram**
[stdatamodels.jwst.datamodels.schema](https://github.com/spacetelescope/stdatamodels/blob/master/src/stdatamodels/jwst/datamodels/schema.py) is an out-of-date duplicate of [stdatamodels.schema](https://github.com/spacetelescope/stdatamodels/blob/master/src/stdatamodels/schema.py). Uses of the stdatamodels.jwst.datamodels.schema were converted to stdatamodels.schema and the automatic shadowing of stdatamodels.jwst.datamodels.schema to jwst.datamodels.schema was removed.

**This removes jwst.datamodels.schema from the api.**

Regression tests run at: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/794/

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by braingram** [comment]
devdeps failures should be fixed by either/both:
https://github.com/spacetelescope/stdatamodels/pull/184
https://github.com/asdf-format/asdf/pull/1594

**Comment by braingram** [comment]
Regression tests show only usual F/D/SFLAT keyword errors. I'm puzzled by these :-/

**Comment by braingram** [comment]
Closing and reopening to trigger CI to pick up stdatamodels changes which should fix devdeps failures.

**Comment by hbushouse** [comment]
> Regression tests show only usual F/D/SFLAT keyword errors. I'm puzzled by these :-/

Aren't we all! Not only are the keywords for those ref files not getting put into the output files, it's obvious that they weren't even applied during the flat-field step, because the data show differences too. Sure would like to figure out what's causing this. It usually comes from the same hand-full of tests, but it's never exactly the same ones on any given regtest run.

**Comment by braingram** [comment]
It's good to know I'm not the only one! :)

I've had numerous fruitless suspicions today including:
- maybe it's something with the test suite execution but...
  - running test in parallel vs serial doesn't seem to matter locally
  - running tests with coverage doesn't seem to matter
  - forcing the order of some of the failing tests doesn't seem to matter (I was combining them with a test that runs the `test_nirspec_verify` `run_image2` fixture which [overrides the f/s/dflats](https://github.com/spacetelescope/jwst/blob/d1a60ee11df7bbeb3b7ea7075ce933b5756a3faa/jwst/regtest/test_nirspec_verify.py#L48C1-L48C1) as 'N/A' (whereas the default value for the argument is None)
- maybe it's something with crds and/or the context (the truth files are 1097 the new files are 1100)

Is there a way to see and/or capture the log output for a test/run on plwishmaster? The inability to replicate the issue locally is making this a hard nut to crack.

**Comment by hbushouse** [comment]
Yes, you can capture the processing log by using the "-s" pytest argument in the test parameters. I've tried this in the past and whenever I do, it all runs just fine (because it knows someone's watching ...). I've also tried the single process vs. multi-processing comparisons by using the "-n 1" pytest option to limit the Jenkins run to only 1 core (instead of many cores). It makes the result WORSE!  Yet more tests then end up with this error. But of course it never happens when running the tests locally on one of our personal systems. So it's highly mind-boggling.

**Comment by braingram** [comment]
Thanks! That is all super useful. I will give some of those options a try.

**Comment by braingram** [comment]
I have hope this will fix the regression test failures: https://github.com/spacetelescope/jwst/pull/7752

**Review by nden — APPROVED** [review]
LGTM

File: CHANGES.rst [hunk 1]
```diff
@@ -65,6 +65,8 @@ datamodels
 - Added two new header keywords to track the rate of cosmic rays and snowball/showers
   [#7609, spacetelescope/stdatamodels [spacetelescope/stdatamodels#173]
 
+- Remove ``jwst.datamodels.schema`` in favor of ``stdatamodels.schema`` [#7660]
```
Comment (hbushouse) [inline CHANGES.rst:68]: This entry will need to be moved to the top of the log, now that it's not part of the recent releases that've happened.

Comment (braingram) [reply CHANGES.rst]: Thanks! I updated the changelog moving this entry to the top, 'unreleased' version.

**Comment by braingram** [comment]
Regtests running at: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/806/

**Review by hbushouse — APPROVED** [review]
Run 806 looks good.

**Pull request merged**